src/Data/ProgrammesDb/EntityRepository/BroadcastRepository.php

```diff
@@ -98,4 +99,48 @@ private function setEntityTypeFilter(QueryBuilder $qb, string $type, string $bro
 
         return $qb;
     }
+
+    public function findAllByServiceAndDateRange(
+        Sid $serviceId,
+        DateTimeImmutable $startDate,
+        DateTimeImmutable $endDate,
+        ?int $limit,
```

Comment:
Convention in the Repositories is that limit and offset must be explicitly stated.
The Repositories are internal classes used the Services (which are the public-facing entry point to the library). The Services have optional pagination parameters